### DISCUSSION TITLE
Allow enums for component properties

### DISCUSSION THREAD

**Discussion by mschop** [livewire/livewire#4411] [3. Feature ideas]
Hi,

I think it would make sense to allow PHP8.1 Enums as the type for component properties.

Example:

```
enum Type: string
{
    case TYPE_A = 'a';
    case TYPE_B = 'b';
}

class MyComponent extends Component
{
   public Type $type;
   ...
}
```

What do you think?

Best Regards
mschop

**Comment by joshhanley** [comment]
@mschop yeah I think that could be useful! You could probably work around it at the moment maybe by using the custom wireable properties on the enum maybe? https://laravel-livewire.com/docs/2.x/properties#wireable-properties

**Comment by andreascreten** [comment]
I started on an implementation here: https://github.com/livewire/livewire/pull/4506
